File: python/bubble_sort.py
def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        print(f"Iterasi {i+1}: {arr}")
        for j in range(0, n-i-1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
    return arr

n = int(input("Jumlah data: "))
data = list(map(int, input("Masukkan data: ").split()))

hasil = bubble_sort(data)
print("Hasil sorting:", hasil)

File: cpp/bubble_sort.cpp
#include <iostream>
using namespace std;

void print(int arr[], int n) {
    for(int i=0;i<n;i++) cout << arr[i] << " ";
    cout << endl;
}

void bubbleSort(int arr[], int n) {
    for(int i=0;i<n;i++) {
        cout << "Iterasi " << i+1 << ": ";
        print(arr, n);

        for(int j=0;j<n-i-1;j++) {
            if(arr[j] > arr[j+1]) {
                swap(arr[j], arr[j+1]);
            }
        }
    }
}

int main() {
    int n;
    cout << "Jumlah data: ";
    cin >> n;

    int arr[n];
    cout << "Masukkan data: ";
    for(int i=0;i<n;i++) cin >> arr[i];

    bubbleSort(arr, n);

    cout << "Hasil sorting: ";
    print(arr, n);
}
